pkg/resources/statefulsets/minio-statefulset.go

```diff
@@ -385,7 +387,8 @@ func NewForMinIO(mi *miniov1.MinIOInstance, serviceName string, hostsTemplate st
 	if mi.Spec.VolumeClaimTemplate != nil {
 		pvClaim := *mi.Spec.VolumeClaimTemplate
 		name := pvClaim.Name
-		for i := 0; i < mi.Spec.VolumesPerServer; i++ {
+		v, _ := strconv.Atoi(mi.Spec.VolumesPerServer)
```

Comment:
Same here, we should validate the string.

Reply:
validation is done initially here, https://github.com/minio/minio-operator/pull/169/files#diff-5fba95502c597c71c1ec5de3290ef01bR453 while someone submits a MinIO Tenant Yaml.